bootstrapper/internal/journald/journald.go

```diff
@@ -31,17 +33,35 @@ func NewCollector(ctx context.Context) (*Collector, error) {
 	if cmd.Err != nil {
 		return nil, cmd.Err
 	}
-	return &Collector{cmd}, nil
+	stdoutPipe, err := cmd.StdoutPipe()
+	if err != nil {
+		return nil, err
+	}
+	stderrPipe, err := cmd.StderrPipe()
+	if err != nil {
+		return nil, err
+	}
+	collector := Collector{cmd, stdoutPipe, stderrPipe}
+	return &collector, nil
+}
+
+// Pipe returns a pipe to read the systemd logs. This should be read with a bufio Reader.
+func (c *Collector) Pipe() (io.ReadCloser, error) {
+	if err := c.cmd.Start(); err != nil {
+		return nil, err
+	}
+	return c.stdoutPipe, nil
 }
 
-// Collect gets all journald logs from a service and returns a byte slice with the plain text logs.
-func (c *Collector) Collect() ([]byte, error) {
-	out, err := c.cmd.Output()
-	var exitErr *exec.ExitError
-	if errors.As(err, &exitErr) {
-		return nil, fmt.Errorf("executing %q failed: %s", c.cmd, exitErr.Stderr)
-	} else if err != nil {
-		return nil, fmt.Errorf("executing command: %w", err)
+// Error returns the error message of the journalctl command.
+// The first two parameters are what's written to stderr as
+// well as the exit/io error, the third one checks if the function
+// ran succesfully.
+func (c *Collector) Error() ([]byte, error, error) {
+	stderr, err := io.ReadAll(c.stderrPipe)
+	if err != nil {
+		return nil, nil, err
 	}
-	return out, nil
+	exitCode := c.cmd.Wait()
+	return stderr, exitCode, nil
 }
```

Comment:
```suggestion
func (c *Collector) Error() ([]byte, error) {
	stderr, err := io.ReadAll(c.stderrPipe)
	if err != nil {
		return nil, err
	}
	return stderr, c.cmd.Wait()
}
```
There is no need to return multiple errors.